### PR TITLE
Travis is dead. Long live Github Actions

### PR DESCRIPTION
**Pull request by hollabaq86**
# why
We haven't been using Travis for some time, but we do need some automated testing setup for PRs.

# what
removes any remaining travis things and adds a basic github actions file. I expect tests to fail with the current setup because integration tests require API credentials.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+on:
+  [push, pull_request]
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+
+    runs-on: ubuntu-20.04 # latest version doesn't have python 3.5
```
Comment (crookedneighbor) [inline .github/workflows/ci.yml:12]: Is this comment needed? Python 3.5 and 3.6 are EOL: https://www.activestate.com/solutions/python-version-support/

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+name: Python
+
+on:
+  [push, pull_request]
```
Comment (crookedneighbor) [inline .github/workflows/ci.yml:4]: this should probably be on pushes to `main` only, otherwise it'll run twice for both pull request and push.

Comment (crookedneighbor) [reply .github/workflows/ci.yml:4]:
```suggestion
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (hollabaq86) [reply .github/workflows/ci.yml:17]: good point, I can drop that comment

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    - name: Setup Test Env
+      run: |
+        pip install -r requirements.txt
+        touch .env
+        echo BT_ENVIRONMENT='${{ secrets.BT_ENVIRONMENT }}' >> .env
+        echo BT_MERCHANT_ID='${{ secrets.BT_MERCHANT_ID }}' >> .env
+        echo BT_PUBLIC_KEY='${{ secrets.BT_PUBLIC_KEY }}' >> .env
+        echo BT_PRIVATE_KEY='${{ secrets.BT_PRIVATE_KEY }}' >> .env
+    - name: Run tests
+      run: python test_app.py
```
Comment (crookedneighbor) [inline .github/workflows/ci.yml:39]: Typically, env variables are done like this:

```suggestion
      env:
        BT_ENVIRONMENT: ${{ secrets.BT_ENVIRONMENT }}
        BT_MERCHANT_ID: ${{ secrets.BT_MERCHANT_ID }}
        BT_PUBLIC_KEY: ${{ secrets.BT_PUBLIC_KEY }}
        BT_PRIVATE_KEY: ${{ secrets.BT_PRIVATE_KEY }}
      run: python test_app.py
```

Comment (hollabaq86) [reply .github/workflows/ci.yml:39]: yeah we'd have to re-write the app to not load env variables from a .env file for this to work

Comment (hollabaq86) [reply .github/workflows/ci.yml:39]: I don't mind doing that, but it changes how folks run their own demo and the scope of these changes. The .env file seems like it historically was an easy way to let devs setup api keys real quick and not have to deal with environment _things_.

**Pull request merged**